components/select_team/index.js

```diff
@@ -4,22 +4,49 @@
 import {connect} from 'react-redux';
 import {bindActionCreators} from 'redux';
 import {withRouter} from 'react-router-dom';
+import {createSelector} from 'reselect';
 
 import {getTeams} from 'mattermost-redux/actions/teams';
 import {loadRolesIfNeeded} from 'mattermost-redux/actions/roles';
 import {getConfig, getLicense} from 'mattermost-redux/selectors/entities/general';
 import {getRoles} from 'mattermost-redux/selectors/entities/roles';
+import {getJoinableTeams, getTeamMemberships} from 'mattermost-redux/selectors/entities/teams';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
 
 import SelectTeam from './select_team.jsx';
 
+const getSortedJoinableTeams = createSelector(
```

Comment:
Maybe it makes sense to move it to mattermost-webapp/selectors, or mattermost-redux. But I'm 1/5 here.

Reply:
PR submitted - https://github.com/mattermost/mattermost-redux/pull/471